File: Web/backend/src/main/java/com/ByTech/ByTech/Payment/controllers/PaymentController.java
package com.ByTech.ByTech.Payment.controllers;

import com.ByTech.ByTech.Payment.models.Payment;
import com.ByTech.ByTech.Payment.repositories.PaymentRepository;
import com.stripe.Stripe;
import com.stripe.model.PaymentIntent;
import com.stripe.param.PaymentIntentCreateParams;
import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/payment")
@CrossOrigin(origins = {"http://localhost:3000","http://localhost:3001"})
public class PaymentController {

    @Autowired
    private PaymentRepository paymentRepository;


    @Value("${stripe.api.key}")
    private String stripeApiKey;

    @PostConstruct
    public void init() {
        Stripe.apiKey = stripeApiKey;
    }

    @PostMapping("/create-payment-intent")
    public Map<String, Object> createPaymentIntent(@RequestBody Map<String,Object> data) throws Exception {

        Long amount = Long.valueOf(data.get("amount").toString());
        String userEmail = data.get("userEmail").toString();

        PaymentIntentCreateParams params = PaymentIntentCreateParams.builder()
                .setAmount(amount)
                .setCurrency("eur")
                .build();

        PaymentIntent intent = PaymentIntent.create(params);

        Map<String, Object> response = new HashMap<>();
        response.put("clientSecret", intent.getClientSecret());
        response.put("paymentId", intent.getId());
        return response;
    }

    @PostMapping("/confirm")
    public ResponseEntity<?> confirmPayment(@RequestBody Map<String, Object> data) {
        String paymentId = data.get("paymentId").toString();
        String userEmail = data.get("userEmail").toString();
        Long amount = Long.valueOf(data.get("amount").toString());

        Payment payment = new Payment();
        payment.setUserEmail(userEmail);
        payment.setAmount(amount);
        payment.setCurrency("eur");
        payment.setStatus("succeeded");
        payment.setCreatedAt(LocalDateTime.now());
        paymentRepository.save(payment);

        return ResponseEntity.ok().build();
    }

    @GetMapping("/all")
    public List<Payment> getAllPayments(){
        return paymentRepository.findAllByOrderByCreatedAtDesc();
    }
}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/Payment/controllers/PaymentController.kt
package com.example.ByTech_movil.Payment.controllers

import com.example.ByTech_movil.Fabricante.repositories.IEnterpriseRepository
import com.example.ByTech_movil.Payment.models.Payment
import com.example.ByTech_movil.Payment.repositories.PaymentRepository
import com.stripe.Stripe
import com.stripe.model.PaymentIntent
import com.stripe.param.PaymentIntentCreateParams
import jakarta.annotation.PostConstruct
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime

@RestController
@RequestMapping("/payment")
@CrossOrigin(origins = ["exp://192.168.0.247:8081"])
class PaymentController {

    @Autowired
    private lateinit var paymentRepository: PaymentRepository

    @Autowired
    private lateinit var enterpriseRepository: IEnterpriseRepository

    @Value("\${stripe.api.key}")
    private lateinit var stripeApiKey: String

    @PostConstruct
    fun init() {
        Stripe.apiKey = stripeApiKey
    }

    @PostMapping("/create-payment-intent")
    @Throws(Exception::class)
    fun createPaymentIntent(@RequestBody data: Map<String, Any>): Map<String, Any> {
        val amount: Long = (data["amount"] as Number).toLong()
        val userEmail: String = data["userEmail"] as String

        val params: PaymentIntentCreateParams = PaymentIntentCreateParams.builder()
            .setAmount(amount)
            .setCurrency("eur")
            .build()

        val intent: PaymentIntent = PaymentIntent.create(params)

        return mapOf(
            "clientSecret" to intent.clientSecret,
            "paymentId" to intent.id
        )
    }

    @PostMapping("/confirm")
    fun confirmPayment(@RequestBody data: Map<String, Any>): ResponseEntity<*> {
        val paymentId: String = data["paymentId"] as String
        val userEmail: String = data["userEmail"] as String
        val amount: Long = (data["amount"] as Number).toLong()

        val payment = Payment().apply {
            this.userEmail = userEmail
            this.amount = amount
            this.currency = "eur"
            this.status = "succeeded"
            this.createdAt = LocalDateTime.now()
        }

        paymentRepository.save(payment)

        return ResponseEntity.ok().build<Any>()
    }

    @GetMapping("/all")
    fun allPayments(): List<Payment> = paymentRepository.findAllByOrderByCreatedAtDesc() ?: emptyList()
}
